src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -226,17 +250,44 @@ def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
                 return [err.error()]
             self.logger.info("partial pileup spec: %s", doc)
 
-        # Based on the document retrieved from the database, apply the user-related
-        # updates and run this new data structure through the usual validation
+        # Based on the document retrieved from the database, apply the user-related updates
         dbDoc.update(doc)
+
+        # add transition record if doc contains new fraction
+        transition = dbDoc.get('transition', [])
+        if not transition:
+            msg = f"To update pileup {pname} document we should already have transition record in place"
+            err = MSPileupUniqueConstrainError(spec, msg)
+            self.logger.error(err)
+            return [err.error()]
+
+        # update transition record if necessary
+        addTransitionRecord(dbDoc, userDN, self.logger)
+        output = self.updatePileupDocumentInDatabase(spec, dbDoc, rseList)
+        return output
+
+    def updatePileupDocumentInDatabase(self, spec, doc, rseList=None, validate=True):
+        """
+        Update given pileup document in database
+        :param spec: MongoDB spec (dictionary) used to locate document in MongoDB
```

Comment:
I would remove `spec` from this list of arguments and construct it here, based on the `doc` object.
This will avoid repetitions at the MSPileupTasks module.

Reply:
ok, done.